src/gpuarray/array.h

```diff
@@ -607,6 +607,78 @@ GPUARRAY_PUBLIC void GpuArray_fprintf(FILE *fd, const GpuArray *a);
 
 GPUARRAY_PUBLIC int GpuArray_fdump(FILE *fd, const GpuArray *a);
 
+/************************************************************************************
+*                         multi-gpu collectives interface                          *
+************************************************************************************/
+
+/**
+ * \brief TODO
+ * \param src [const GpuArray*] TODO
+ * \param dest [GpuArray*] TODO
+ * \param count [int] TODO
+ * \param typecode [int] TODO
+ * \param opcode [int] TODO
+ * \param root [int] TODO
+ * \param comm [gpucomm*] TODO
+ * \return int TODO
+ */
+GPUARRAY_PUBLIC int GpuArray_reduce(const GpuArray* src, GpuArray* dest,
+                                    int count, int typecode, int opcode,
+                                    int root, gpucomm* comm);
```

Comment:
I think I got this. The design with which you developed this. It would be good to write documentation about this. Question: Why did you choose C over C++? :)